plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -119,7 +119,7 @@ private static void sendPostRequest(String parametersString) {
       connection = (HttpURLConnection) url.openConnection();
       connection.setDoOutput(true);
       connection.setRequestMethod("POST");
-      connection.setFixedLengthStreamingMode(parametersString.length());
+      connection.setFixedLengthStreamingMode(parametersString.getBytes("UTF-8").length);
```

Comment:
You could save the bytes and write it out directly into the `connection.getOutputStream()` rather than re-encoding it again. No doubt this will reap huge performance gains!

Otherwise LGTM.